src/diffusers/schedulers/scheduling_euler_discrete.py

```diff
@@ -191,7 +200,16 @@ def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.devic
         """
         self.num_inference_steps = num_inference_steps
 
-        timesteps = np.linspace(0, self.config.num_train_timesteps - 1, num_inference_steps, dtype=float)[::-1].copy()
+        # "linspace" and "leading" corresponds to annotation of Table 1. of https://arxiv.org/abs/2305.08891
+        if self.config.timestep_spacing == "linspace":
+            timesteps = np.linspace(0, self.config.num_train_timesteps - 1, num_inference_steps, dtype=float)[::-1].copy()
+        elif self.config.timestep_spacing == "leading":
+            step_ratio = self.config.num_train_timesteps // self.num_inference_steps
```

Comment:
this new spacing, doesn't give drastically better results, but better results nevertheless IMO. It's also needed to have 1-to-1 the same results as original code.

Reply:
Does the original code (XL) use this new spacing scheme, though?